.github/workflows/bcny-cmake.yml

```diff
@@ -6,44 +6,88 @@ on:
 jobs:
   build:
     runs-on: swift-build-windows-latest-32-cores
+    outputs:
+      artifact_amd64: ${{ steps.vars.outputs.artifact_amd64 }}
+      artifact_arm64: ${{ steps.vars.outputs.artifact_arm64 }}
+      release: $${{ steps.vars.outputs.release }}
+
+    strategy:
+      matrix:
+        include:
+          - arch: 'amd64'
+            cmake_arch: 'AMD64'
+            cpack_arch: 'x64'
+          - arch: 'arm64'
+            cmake_arch: 'ARM64'
+            cpack_arch: 'arm64'
 
     steps:
     - uses: compnerd/gha-setup-vsdevenv@main
+      with:
+        host_arch: amd64
+        components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
+        arch: ${{ matrix.arch }}
     - uses: actions/checkout@v4
       with:
         path: ${{ github.workspace }}/SourceCache/cmake
-    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release
-    - run: cmake --build ${{ github.workspace }}/BinaryCache/cmake
+    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake/${{ matrix.arch }} -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release -D CMAKE_SYSTEM_NAME=Windows -D CMAKE_SYSTEM_PROCESSOR=${{ matrix.cmake_arch }} -D CMAKE_CROSSCOMPILING=1
```

Comment:
Each step should run in its own parallel container, so we don't need to change the output directory (`-B`).

If we set `CMAKE_SYSTEM_NAME` and `CMAKE_SYSTEM_PROCESSOR`, `CMAKE_CROSSCOMPILING` is implied I thought, does that not work?

Might be nice to split this up at this point into multiline command.

Reply:
Done